core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -1135,6 +1135,73 @@ public boolean hasNext() {
     return Streams.stream(rollingIterator(n));
   }
 
+  /**
+   * Transposes data in the table, switching rows for columns. For example, a table like this.<br>
+   * label | value1 | value2 |<br>
+   * -------------------------------<br>
+   * row1 | 1 | 2 |<br>
+   * row2 | 1.1 | 2.1 |<br>
+   * row3 | 1.2 | 2.2 |<br>
+   *
+   * <p>Is transposed into the following<br>
+   * label | row1 | row2 | row3 |<br>
+   * -------------------------------------<br>
+   * value1 | 1 | 1.1 | 1.2 |<br>
+   * value2 | 2 | 2.1 | 2.2 |<br>
+   *
+   * @return The transposed table
+   */
+  public Table transpose() {
+    Table transposed = Table.create(this.name);
+    if(this.columnCount() == 0)
+    {
+      return this;
+    }
+
+    ColumnType[] types = this.columnTypes();
+    // If all columns are of the same type
+    long distinctColumnTypesCount = Arrays.stream(types).skip(1).distinct().count();
+    if (distinctColumnTypesCount > 1) {
+      throw new IllegalArgumentException(
+          "Transpose currently only supports tables where value columns are of the same type");
+    }
+
+    ColumnType resultType = types[1];
+    StringColumn labelColumn = StringColumn.create(this.column(0).name());
+    transposed.addColumns(labelColumn);
+    for (String label : this.column(0).asStringColumn()) {
+      transposed.addColumns(resultType.create(label));
+    }
+
+    for (int i = 1; i < this.columnCount(); i++) {
+      Column<?> columnToTranspose = this.column(i);
+      labelColumn.append(columnToTranspose.name());
+    }
+
+    for (int i = 1; i < transposed.columnCount(); i++) {
+      Column<?> column = transposed.column(i);
+      int row = i - 1;
+      for (int col = 1; col < this.columnCount(); col++) {
+        // Avoid boxing for primitives
```

Comment:
I considered doing it that way, but I think it will be very inefficient memory/performance wise.
Transposing a whole table is a big operation, so I think that matters.

Reply:
I'm not following. Why would it use more memory or have worse performance?

Reply:
Having looked at this more closely, I take back that comment. I'd tried to do something previously using existing methods but it would have resized the data structure inside the columns when it appended, which would be inefficient.

Do you already have other cases in mind that could reuse that method ? I'd typically take the view that until it is needed elsewhere it shouldn't be on the public API, but I do agree it might be useful there.

I'd suggest that it might be better to name it appendFrom.

Reply:
That's fair. I'm fine leaving this internal for now. We do something somewhat similar in `DataFrameJoiner`, but it looks different enough that it couldn't easily be shared